pkg/controllers/updaterun/controller.go

```diff
@@ -312,6 +317,34 @@ func handleClusterApprovalRequest(oldObj, newObj client.Object, q workqueue.Type
 	})
 }
 
+// handleClusterApprovalRequestDelete finds the ClusterStagedUpdateRun creating the ClusterApprovalRequest,
+// and enqueues it to the ClusterStagedUpdateRun controller queue when the ClusterApprovalRequest is deleted.
+func handleClusterApprovalRequestDelete(obj client.Object, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	appReq, ok := obj.(*placementv1beta1.ClusterApprovalRequest)
+	if !ok {
+		klog.V(2).ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("cannot cast runtime object to ClusterApprovalRequest")),
+			"Invalid object type", "object", klog.KObj(obj))
+		return
+	}
+	isApproved := meta.FindStatusCondition(appReq.Status.Conditions, string(placementv1beta1.ApprovalRequestConditionApproved))
+	approvalAccepted := condition.IsConditionStatusTrue(meta.FindStatusCondition(appReq.Status.Conditions, string(placementv1beta1.ApprovalRequestConditionApprovalAccepted)), appReq.Generation)
+	if isApproved != nil && approvalAccepted {
```

Comment:
Why do we need to check if "isApproved != nil"?

Reply:
It's a strict check to ensure ApprovalRequest is approved with either true/false(as setting it to false is ignored once approvalAccepted condition is added) and then approvalAccepted condition is set to true